File: language-support/js/daml-ledger/src/validation/submit_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { object, Validation } from "./base";
import { RequiredValidation } from "./base/typelevel";
import { Commands } from "./commands";

function required(): RequiredValidation<ledger.SubmitRequest> {
    return {
        commands: Commands
    };
}

export const SubmitRequest: Validation = object<ledger.SubmitRequest>('SubmitRequest', required, () => ({}));

File: canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/SubmitRequest.java
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates.
// Proprietary code. All rights reserved.

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v2.CommandSubmissionServiceOuterClass;
import org.checkerframework.checker.nullness.qual.NonNull;

public final class SubmitRequest {

  public static CommandSubmissionServiceOuterClass.SubmitRequest toProto(
      @NonNull CommandsSubmission submission) {
    return CommandSubmissionServiceOuterClass.SubmitRequest.newBuilder()
        .setCommands(submission.toProto())
        .build();
  }

  public static CommandsSubmission fromProto(
      CommandSubmissionServiceOuterClass.@NonNull SubmitRequest request) {
    return CommandsSubmission.fromProto(request.getCommands());
  }
}
